frontend/src/components/Toolbar/Toolbar.tsx

```diff
@@ -3,43 +3,50 @@
  * @author jhuebner
  * @date 06.09.2021
  */
+
 import React, { useState } from 'react';
+import { connect } from 'react-redux';
 import { Link } from 'react-router-dom';
 import { FaBars } from 'react-icons/fa';
 import Logo from '../../images/kant.png';
 import useWindowDimensions from '../../hooks/useWindowDimensions';
 import './Toolbar.css';
 import { LoginLogoutButton } from './LoginLogoutButton/LoginLogoutButton';
 
-const links = [
-  {
-    id: 1,
-    name: 'AG-Buchung',
-    path: '/student/ag',
-  },
-  {
-    id: 2,
-    name: 'Facharbeit',
-    path: '/student/facharbeit',
-  },
-  {
-    id: 3,
-    name: 'Wahlpflicht',
-    path: '/student/wahlpflicht',
-  },
-  {
-    id: 4,
-    name: '5. PK',
-    path: '/student/pruefungskomponente',
-  },
-  {
-    id: 5,
-    name: 'Einstellungen',
-    path: '/settings',
-  },
-];
-
-function Toolbar() {
+function Toolbar(props) {
+  const { isLoggedIn, role, accessibleComponents } = props;
+  let approprateRolePath = '';
+  if (role === 'student') {
+    approprateRolePath = 'student';
+  } else if (role === 'admin') {
+    approprateRolePath = 'admin';
+  }
+  const links = [
+    {
+      id: 1,
+      name: 'AG-Buchung',
+      path: `/${approprateRolePath}/ag`,
+      disabled: (!(accessibleComponents.includes('ag')) && role === 'admin') || !isLoggedIn,
+    },
+    {
+      id: 2,
+      name: 'Facharbeit',
+      path: `/${approprateRolePath}/facharbeit`,
+      disabled: (!(accessibleComponents.includes('fa')) && role === 'admin') || !isLoggedIn,
```

Comment:
wieso genau die Abhängigkeit von der rolle, ob der Button in der Toolbar disabled ist?

Reply:
hat sich erledigt, schon klar: Für einen Admin soll dass ja gar nicht deaktiviert sein, damit er die auch wieder aktivieren kann :D